kuber/utils.go

```diff
@@ -40,6 +42,8 @@ func (s *ParentsStore) SetParents(namespace string, kind string, name string, pa
 }
 
 func (s *ParentsStore) GetParents(namespace string, kind string, name string) (*ParentController, bool) {
+	s.Lock()
+	defer s.Unlock()
 	parents, found := s.parents[GetEntityKey(namespace, kind, name)]
```

Comment:
Why lock on reading?

Reply:
It appeared as a data race when I ran the build with `-race` flag as if you're going to read a variable that might be set from another goroutine, then you need to lock the associated mutex before you read the value to ensure that you actually see the updates

Reply:
but I think you're right since all the writes to that map have locks on them. I wonder what triggered the data race though

Reply:
I removed it and will inspect the data race when/if it showed up again